Makefile

```diff
@@ -321,6 +321,16 @@ start-upgrade-test-admin: zetanode-upgrade
 	export E2E_ARGS="--skip-regular --test-admin" && \
 	cd contrib/localnet/ && $(DOCKER_COMPOSE) --profile upgrade -f docker-compose.yml -f docker-compose-upgrade.yml up -d
 
+# this test upgrades from v18 and execute the v2 contracts migration process
+# this tests is part of upgrade test part because it should run the upgrade from v18 to fully replicate the upgrade process
+start-upgrade-v2-migration-test: zetanode-upgrade
+	@echo "--> Starting v2 migration upgrade test"
+	export LOCALNET_MODE=upgrade && \
+	export UPGRADE_HEIGHT=90 && \
+	export E2E_ARGS="--test-v2-migration" && \
+	cd contrib/localnet/ && $(DOCKER_COMPOSE) --profile upgrade -f docker-compose.yml -f docker-compose-upgrade.yml up -d
```

Comment:
I think so yeah, --profile parametrize what is used in the image afaik

Reply:
Yes, `--profile` enables the upgrade orchestrator components, `docker-compose-upgrade.yml` overwrites the image versions

Putting all our containers in a single docker compose then using `--profile` allows us to avoid 1) IP conflicts and 2) mistakes when refactoring (it's more obvious what you need to update when it's one file)